metricbeat/module/windows/service/reader.go

```diff
@@ -0,0 +1,198 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build windows
+
+package service
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"strconv"
+	"syscall"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows/registry"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+)
+
+var (
+	// errorNames is mapping of errno values to names.
+	// https://msdn.microsoft.com/en-us/library/windows/desktop/ms681383(v=vs.85).aspx
+	errorNames = map[uint32]string{
+		1077: "ERROR_SERVICE_NEVER_STARTED",
```

Comment:
not sure why there is only one, this could be the most common one but I would either add several more or strip this entire logic all together and just retrieve the error code. I would go for the latter, what do you think?

Reply:
Sounds good to me. As it comes from the existing code we can also leave this for a separated change.